Add initial support for  _resultDependsOnSelf

meg-gupta opened the pull request:
These attributes are used to establish lifetime dependence between argument/self and the result.

Matching swift-syntax PR :  https://github.com/apple/swift-syntax/pull/2281

ahoppen reviewed (commented):
This change will need a corresponding PR in swift-syntax to add support for parsing `_resultDependsOn` in the new parser as well.

meg-gupta commented:
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

hborla commented:
Please gate the implementation of a underscored attributes behind an experimental feature flag; see the `EXPERIMENTAL_FEATURE` macro in Features.def for examples.

atrick commented:
I'll add the NonEscapableTypes experimental feature flag in this PR: https://github.com/apple/swift/pull/69164

meg-gupta commented:
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

meg-gupta commented:
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

atrick commented:
PR https://github.com/apple/swift/pull/69164 landed. You can use the experimental feature flag here.

atrick commented on test/Sema/result_depends_on_errors.swift:
```diff
@@ -0,0 +1,15 @@
+// RUN: %target-typecheck-verify-swift  -enable-builtin-module
```
Hopefully this test now fails without `-enable-experimental-feature NonEscapableTypes`.
Then you'll need `REQUIRES: asserts`

meg-gupta commented:
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

meg-gupta commented:
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

meg-gupta commented:
Previous run of tests passed, but main has changed, so testing again..

meg-gupta commented:
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test

meg-gupta committed:
Add initial support for _resultDependsOnSelf

This is used to establish lifetime dependence between self and the result.

Add under NonEscapableTypes experimental feature

meg-gupta commented:
Please test with following pull request:
https://github.com/apple/swift-syntax/pull/2281

@swift-ci Please test
